src/elements/components/FeatheryAuthGate.tsx

```diff
@@ -0,0 +1,178 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import { Form } from '../..';
+import { Props as FormProps } from '../../Form';
+import { getStytchJwt } from '../../utils/browser';
+import { getAuthClient, initInfo, initState } from '../../utils/init';
+import { authHookCb } from '../../integrations/stytch';
+import Client from '../../utils/client';
+import { isAuthStytch } from '../../integrations/utils';
+import Spinner from './Spinner';
+import { isHrefFirebaseMagicLink } from '../../integrations/firebase';
+/** TODO: These next 2 should maybe be dynamically imported, but having trouble with that
+ * combined 6.9k gzipped, so OK for now
+ */
+import { useIdleTimer } from 'react-idle-timer';
+import throttle from 'lodash.throttle';
+
+const TEN_SECONDS_IN_MILLISECONDS = 1000 * 10;
+const FIVE_MINUTES_IN_MILLISECONDS = 1000 * 60 * 5;
+
+export const authState = {
+  // This is a flag so we only redirect to the login start step immediately
+  // after auth, not during other form navigation
+  redirectAfterLogin: false,
+  sentAuth: false,
+  clearLoader: () => {},
+  onLogin: () => {},
+  onLogout: () => {}
+};
+
+const FeatheryAuthGate = ({
+  authId: authIdProp,
+  formProps,
+  whitelistPath,
+  fullPageLogin,
+  onLogin = () => {},
+  onLogout = () => {},
+  children
+}: {
+  authId?: string;
+  formProps: FormProps;
+  whitelistPath?: string;
+  fullPageLogin?: boolean;
+  onLogin?: () => void;
+  onLogout?: () => void;
+  children: JSX.Element;
+}) => {
+  const formCompleted = initInfo().sessions[formProps.formName]?.form_completed;
+
+  // Need to use this flag because when doing magic link login the onChange
+  // event doesn't seem to be added early enough to catch the first event which
+  // is the one containing the token. subsequent events do not contain the token
+  const hasAuthedRef = useRef(false);
+  // Use this render state to force re-evaluation of initState, since initState isn't reactive as-is
```

Comment:
what initState use case are you trying to refresh for?

Reply:
updated comment to reflect the updated code. don't think I could get rid of this without further changes because authId is still not react/redux state